PowerPointLabs/PowerPointLabs/SyncLab/SyncLabShapeStorage.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System;
+
+using Microsoft.Office.Interop.PowerPoint;
+
+using PowerPointLabs.Models;
+
+namespace PowerPointLabs.SyncLab
+{
+    public class SyncLabShapeStorage : PowerPointPresentation
+    {
+#pragma warning disable 0618
+
+        private int nextKey = 0;
+
+        public SyncLabShapeStorage() : base()
+        {
+            Path = Globals.ThisAddIn.PrepareTempFolder(
```

Comment:
`Globals.ThisAddIn.PrepareTempFolder` can be refactored into a kind of StorageService.
`PowerPointPresentation.Current.Presentation` here I guess it's just `this` presentation.